tests/functional/build/fixtures.py

```diff
@@ -0,0 +1,474 @@
+"AND","Andorra","AD","20","AND","232","6","20","ADO","","AN","Andorra","3041565","ANDORRA","","","","","","","",""
+"ANT","Netherlands Antilles","AN","530","","","","","ANT","353","NT","Netherlands Antilles","","NETHERLANDS ANTILLES","Netherlands Antilles","211","Netherlands Antilles","361","Netherlands Antilles","ANT","AN","530"
+"ARE","United Arab Emirates","AE","784","UAE","696","225","784","ARE","466","AE","United Arab Emirates","290557","UNITED ARAB EMIRATES","United Arab Emirates","140","United Arab Emirates","576","","","","""
+    ""
+)
+
+models_circular_relationship__test_yml = """
+version: 2
+
+models:
+  - name: model_0
+    columns:
+      - name: iso3
+        tests:
+          - relationships:
+              to: ref('model_1')
+              field: iso3
+
+  - name: model_1
+    columns:
+      - name: iso3
+        tests:
+          - relationships:
+              to: ref('model_0')
+              field: iso3
+
+"""
+
+models_circular_relationship__model_0_sql = """
+{{ config(materialized='table') }}
+
+select * from {{ ref('countries') }}
+"""
+
+models_circular_relationship__model_1_sql = """
+{{ config(materialized='table') }}
+
+select * from {{ ref('model_0') }}
+
+"""
+
+models_circular_relationship__model_99_sql = """
+{{ config(materialized='table') }}
+
+select '1' as "num"
+
+
+"""
+
+models_simple_blocking__schema_yml = """
+version: 2
+
+models:
+  - name: model_a
+    columns:
+      - name: id
+        tests:
+          - not_null
+
+"""
+
+models_simple_blocking__model_b_sql = """
+select * from {{ ref('model_a') }}
+
+"""
+
+models_simple_blocking__model_a_sql = """
+select null as id
+
+"""
+
+test_files__model_b_sql = """
+select * from {{ ref('model_a') }}
+
+"""
+
+test_files__model_b_null_sql = """
+select null from {{ ref('model_a') }}
+
+"""
+
+models_interdependent__schema_yml = """
+version: 2
+
+models:
+  - name: model_a
+    columns:
+      - name: id
+        tests:
+          - unique
+          - not_null
+          - relationships:
+              to: ref('model_b')
+              field: id
+          - relationships:
+              to: ref('model_c')
+              field: id
+
+  - name: model_b
+    columns:
+      - name: id
+        tests:
+          - unique
+          - not_null
+          - relationships:
+              to: ref('model_a')
+              field: id
+          - relationships:
+              to: ref('model_c')
+              field: id
+
+  - name: model_c
+    columns:
+      - name: id
+        tests:
+          - unique
+          - not_null
+          - relationships:
+              to: ref('model_a')
+              field: id
+          - relationships:
+              to: ref('model_b')
+              field: id
+
+"""
+
+models_interdependent__model_c_sql = """
+select * from {{ ref('model_b') }}
+
+"""
+
+models_interdependent__model_a_sql = """
+select 1 as id
+
+"""
+
+
```

Comment:
Everything below this comment should be defined in the test file.  It makes it much more clear what's happening in the tests that way.  The project fixture takes care of writing all the files to the project.

Reply:
I've been putting the actual model strings into the test files most of the time.  I tend to only have strings in this file that are extremely long or shared across multiple files.  If it makes sense to do so i also split up the single giant test file into multiple files to make it easier to find and organize tests.  That doesn't always make sense though, it's a different call for each set of tests, not a rule.

Reply:
> Everything below this comment should be defined in the test file. It makes it much more clear what's happening in the tests that way. 

I'm not sure I agree with that assessment.  IMO it's easier to look at the test logic in one place (the test file) and the underlying project data in another file (the fixtures file).  That said, I don't have strong feelings here so I'm happy to move it if it's more consistent.

Reply:
I hear you.  I'm going to start a thread so we can decide on a standard here as a team.